addons/popochiu/engine/interfaces/i_room.gd

```diff
@@ -110,13 +117,197 @@ func clear_instances() -> void:
 	_room_instances.clear()
 
 
+## Loads the room with [param script_name]. [param use_transition] can be used to trigger a [i]fade
+## out[/i] animation before loading the room, and a [i]fade in[/i] animation once it is ready.
+## If [param store_state] is [code]true[/code] the state of the room will be stored in memory.
+## [param ignore_change] is used internally by Popochiu to know if it's the first time the room is
+## loaded when starting the game.
+func goto_room(
+	script_name := "",
+	use_transition := true,
+	store_state := true,
+	ignore_change := false
+) -> void:
+	if not E.in_room: return
+	
+	E.in_room = false
+	G.block()
+	
+	_use_transition_on_room_change = use_transition
+	if use_transition:
+		E.tl.play_transition(E.tl.FADE_IN)
+		await E.tl.transition_finished
+	
+	# Prevent the GUI to show info from the previous room
+	G.show_hover_text()
+	Cursor.show_cursor()
+	
+	if is_instance_valid(C.player) and Engine.get_process_frames() > 0:
+		C.player.last_room = current.script_name
+	
+	# Store the room state
+	if store_state:
+		rooms_states[current.script_name] = current.state
+		current.state.save_childs_states()
```

Comment:
this method should be called `save_children_states()`

Reply:
Done.